src/azure-cli-core/azure/cli/core/__init__.py

```diff
@@ -289,6 +289,11 @@ def __init__(self, cli_ctx=None, min_profile=None, max_profile='latest',
         self.skip_applicability = False
         self._command_group_cls = command_group_cls or AzCommandGroup
         self._argument_context_cls = argument_context_cls or AzArgumentContext
+        if custom_resource_types:
+            from azure.cli.core.profiles import API_PROFILES
+            for profile, v in custom_resource_types.items():
+                if profile in API_PROFILES:
+                    API_PROFILES[profile].update(v)
```

Comment:
Instead of here, how about adding this update as a callable function in azure.cli.core.profiles?

Reply:
I've gone with that approach.